index.bs

```diff
@@ -7255,272 +7268,297 @@ The weight that [=[RPS]=] give to the presence of a signature from a [=device-bo
+Note that when |nonce| is empty, then the (signed) authenticator extension output MAY be constant. However, the (unsigned) |spkSig| output is always unique and prevents replay of the (signed) extension output without knowledge of the supplemental private key.
 
-#### `devicePubKey` Extension Output Verification Procedures #### {#sctn-device-publickey-extension-verification}
+#### `supplementalPubKeys` Extension Output Verification Procedures #### {#sctn-supplemental-public-keys-extension-verification}
 
-Verifying the <code>[=devicePubKey=]</code> extension output is performed by the [=[RP]=] whenever a [=device public key=] is returned within the extension output.
+Verifying the <code>[=supplementalPubKeys=]</code> extension output is performed by the [=[RP]=] whenever a supplemental public key is returned within the extension output.
 
-The [=devicePubKey=] extension adds the following [=struct/item=] to [=credential records=]:
+The [=supplementalPubKeys=] extension adds the following [=struct/item=] to [=credential records=]:
 
 <dl dfn-for="credential record" dfn-type="abstract-op">
-    :   <dfn>devicePubKeys</dfn>
-    ::  An initially [=set/empty=] [=set=] of [=device-bound key records=] associated with this [=public key credential source=].
+    :   <dfn>supplementalPubKeys</dfn>
+    ::  An initially [=set/empty=] [=set=] of [=supplemental public key records=] associated with this [=public key credential source=].
 
-        A <dfn dfn>device-bound key record</dfn> is an abstract representation of a registered [=device-bound key=].
+        A <dfn dfn>supplemental public key record</dfn> is an abstract representation of a registered supplemental public key.
         It is a [=struct=] with the following [=struct/items=]:
 
-        <dl dfn-for="devicePubKey record" dfn-type="abstract-op">
+        <dl dfn-for="supplementalPubKeys record" dfn-type="abstract-op">
             :   <dfn>aaguid</dfn>
-            ::  The [=AAGUID=] of the [=device-bound key=]'s [=managing authenticator=].
+            ::  The [=AAGUID=] included with the supplemental public key.
                 This MAY be different from the [=AAGUID=] in the [$credential record/attestationObject$], if any, of the containing [=credential record=].
 
-            :   <dfn>dpk</dfn>
-            ::  The public key portion of the [=device-bound key=].
+            :   <dfn>spk</dfn>
+            ::  The public key portion of the supplemental public key.
 
             :   <dfn>scope</dfn>
-            ::  The scope of the [=device-bound key=]. See [[#sctn-device-publickey-extension-definition]] for details.
+            ::  The scope of the supplemental public key. See [[#sctn-supplemental-public-keys-extension-definition]] for details.
```

Comment:
This doesn't map one-to-one to the `attObjForSupplementalPublicKey` structure anymore, since that now has no `scope` key but instead may contain either a `provider` or a `device` key. What is the benefit of defining `scope` as a group with different choices of key, rather than a single `scope` value with an enumeration value?

Reply:
An authenticator can only return one SPK per scope, but "device-wide" and "per-app" are flavours of the same scope. I agree that this is awkward. At this point I don't believe that anyone plans to implement the "per-app" flavour of device-scoped SPKs and so I've removed it. Hopefully that makes things easier.

Reply:
Ah, I see. I hadn't picked up on that restriction, but I see now that this is implied `For each |scope| in |scopes|:` in the authenticator processing.

Sure, if we don't need the "per-app" scope I'm happy to axe it.